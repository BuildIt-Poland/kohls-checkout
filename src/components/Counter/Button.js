import styled from 'styled-components';

const Button = styled.button`
  font-size: 14px;
  background-color: #00a09f;
  color: #ffffff;
`;

export default Button;
