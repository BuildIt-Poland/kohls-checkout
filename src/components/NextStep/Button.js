import styled from 'styled-components';
import {
  colorTurquoise,
  colorWhite,
  colorYellow,
  colorLightGrey,
  fontWeightBold,
  fontLarge,
  borderWidthXThick
} from '../../styles/designTokens';

const Button = styled.button`
  background: ${colorTurquoise};
  border: 0;
  color: ${colorWhite};
  display: block;
  font-size: ${fontLarge};
  font-weight: ${fontWeightBold};
  padding: 1.2rem;
  text-align: center;
  text-decoration: none;
  width: 100%;
  max-width: 96rem;
  margin: 0 auto;

  &:not(:disabled) {
    cursor: pointer;
  }

  &:disabled {
    background-color: ${colorLightGrey};
  }

  &:focus {
    outline: ${borderWidthXThick} solid ${colorYellow};
    outline-offset: 0;
  }
`;

export default Button;
