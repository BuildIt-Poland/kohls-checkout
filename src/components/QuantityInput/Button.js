import styled from 'styled-components';
import {
  colorBlack,
  colorWhite,
  colorYellow,
  colorLightGrey,
  fontMedium,
  fontWeightBold,
  borderWidthXThick
} from '../../styles/designTokens';

const Button = styled.button`
  width: 4rem;
  height: 4rem;
  border: 0;
  background: ${colorBlack};
  color: ${colorWhite};
  font-size: ${fontMedium};
  font-weight: ${fontWeightBold};
  display: flex;
  align-items: center;
  justify-content: center;

  &:disabled {
    background: ${colorLightGrey};
  }

  &:not(:disabled) {
    cursor: pointer;
  }

  &:focus {
    outline: ${borderWidthXThick} solid ${colorYellow};
    outline-offset: 0;
  }
`;

export default Button;
