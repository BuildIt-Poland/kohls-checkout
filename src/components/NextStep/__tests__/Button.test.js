import React from 'react';
import { create } from 'react-test-renderer';

import Button from '../Button';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - NextStep Button', () => {
  it('render NextStep component', () => {
    const component = create(<Button>Test</Button>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
