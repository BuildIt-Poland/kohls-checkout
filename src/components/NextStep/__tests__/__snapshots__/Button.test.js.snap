// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - NextStep Button render NextStep component 1`] = `
.c0 {
  background: #00a09f;
  border: 0;
  color: #ffffff;
  display: block;
  font-size: 2.4rem;
  font-weight: 700;
  padding: 1.2rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  max-width: 96rem;
  margin: 0 auto;
  cursor: pointer;
}

.c0:disabled {
  background-color: #b0b0b0;
}

<button
  className="c0"
>
  Test
</button>
`;
